lib/util/trip/trip_page_builder.dart

```diff
@@ -38,31 +35,47 @@ class TripPageBuilder {
             children: <Widget>[
               TripStreamBuilder<T>(
                 stream: trips,
-                emptyMessage: S.of(context).widgetTripBuilderNoUpcoming(name),
-                filterTrips: (List<T> trips) => trips
-                    .where((T trip) =>
-                        trip.endTime.isAfter(DateTime.now()) &&
-                        (trip is! Ride || trip.status != RideStatus.withdrawnByRider) &&
-                        !trip.hideInListView)
-                    .toList(),
-                tripCard: tripCard,
+                emptyMessage: isRide
+                    ? S.of(context).widgetTripBuilderNoUpcomingRides
+                    : S.of(context).widgetTripBuilderNoUpcomingDrives,
+                filterTrips: (List<T> trips) =>
+                    trips.where((T trip) => trip.shouldShowInListView(past: false)).toList(),
               ),
               TripStreamBuilder<T>(
                 stream: trips,
-                emptyMessage: S.of(context).widgetTripBuilderNoPast(name),
-                filterTrips: (List<T> trips) => trips.reversed
-                    .where((T trip) =>
-                        trip.endTime.isBefore(DateTime.now()) &&
-                        (trip is! Ride || trip.status != RideStatus.withdrawnByRider) &&
-                        !trip.hideInListView)
-                    .toList(),
-                tripCard: tripCard,
+                emptyMessage:
+                    isRide ? S.of(context).widgetTripBuilderNoPastRides : S.of(context).widgetTripBuilderNoPastDrives,
+                filterTrips: (List<T> trips) =>
+                    trips.reversed.where((T trip) => trip.shouldShowInListView(past: true)).toList(),
               ),
             ],
           ),
         ),
-        floatingActionButton: Semantics(sortKey: const OrdinalSortKey(0), child: floatingActionButton),
+        floatingActionButton: Semantics(
+          sortKey: const OrdinalSortKey(0),
+          child: getFloatingActionButton(isRide, context),
+        ),
       ),
     );
   }
+
+  Widget getFloatingActionButton(bool isRide, BuildContext context) {
+    if (isRide) {
```

Comment:
I think it would be better if you used the `?` operator or variables. `onPressed` and `backgroundColor` are still the same, and I think it should be visible that this is logically the same button

Reply:
I thought about that too. It's ride on the edge where 3 of 5 arguments are different lol.